gprofiler/spark/sampler.py

```diff
@@ -47,204 +39,19 @@ def __init__(
         self._sample_period = sample_period
         # not the same instance as GProfiler._stop_event
         self._stop_event = Event()
-        self._spark_sampler: Optional[SparkCollector] = None
+        self._spark_sampler: Optional[BigDataSampler] = None
```

Comment:
Any reason you initialize it to None here then override it 5 lines later? Remove this line?